### DISCUSSION TITLE
in-game tablet not detecting and tablet mess

### DISCUSSION THREAD

**Discussion by amchoudhry** [ppy/osu#13807] [Q&A] [answered]
I've been looking around and I cant find anyone who has the same issue as me. on osu!stable, my gaomon s620 works completely fine and has no issues at all. I want to get into lazer since its the future of osu! but my table is not working correctly. I have a video here showcasing my problem. I can use my tablet completely fine on the menu and stuff. but as soon as i get into the game and pick up my pen and put it near the tablet, the cursor goes all the way on the right side of the screen. also my tablet is not being detected. ive even reinstalled my gaomon tablet driver and it didnt do anything. i tried windowed mode, borderless, fullscreen, tried reinstalling lazer and my tablet is still not working. i even turned off my tablet driver and it worked but it felt like i was playing on a grid and didnt feel like how my tablet feels with stable. the clip below is me using my tablet driver with borderless just so you know. i pick up my pen at 0:28

https://user-images.githubusercontent.com/87052010/124694054-8dd2a480-deae-11eb-9706-fb0f437957f8.mp4

 i want to solve this because i want to play lazer!!! help me out :)

**Comment by peppy** [comment, marked as answer]
Whatever "driver" or tablet software you have is not handling mouse confining. You can disable this as a temporary solution (search for "confine" in settings and set it to "never")

Could you confirm whether you have similar issues in osu-stable if you run it windowed with "confine" set to always? That would help us figure out why you're not experiencing the same issues on stable.

**Reply by amchoudhry** [reply]
1. i set osu!lazer into windowed mode and had confine set to never and it works but its not picking up my tablet driver area and it feels very wonky to play on. my tablet is still not being detected. i also tried lazer with fullscreen where the mouse confining cant be changed and my cursor is stuck in the top right corner and is not visible when ever i put my pen near the tablet. but as soon as i get into game mode, it has the same issue as the video above. also thought i should give you this:

![tablet driver ss4](https://user-images.githubusercontent.com/87052010/124794072-a24d8600-df1c-11eb-9631-d7e93bace7f0.PNG)
![tablet driver ss3](https://user-images.githubusercontent.com/87052010/124794073-a24d8600-df1c-11eb-9c21-d4ce5f7c9c05.PNG)
![tablet driver ss2](https://user-images.githubusercontent.com/87052010/124794075-a24d8600-df1c-11eb-9bfd-f3f9a69e3d02.PNG)
![tablet driver ss1](https://user-images.githubusercontent.com/87052010/124794077-a2e61c80-df1c-11eb-806c-47796851ab21.PNG)

2. i set osu!stable into windowed mode and had confine set to always and no issue except tablet latency.

**Reply by peppy** [reply]
I'm not sure. The only thing I can think of that changes during gameplay is the confine mode.

**Reply by amchoudhry** [reply]
if i figure anything out, ill make sure to post it here. anyways thanks for the help :)

**Reply by peppy** [reply]
The only other edge case is high precision input. make sure to test with that on / off. But that shouldn't change during gameplay.

**Reply by amchoudhry** [reply]
figured out a semi solution, turning off raw input / high mouse precision. but its weird. i changed my sensitivity in the tablet driver and it didnt work on lazer or stable. and my tablet is still not being detected :(

**Reply by frenzibyte** [reply]
A combination of mouse mode confinement and high mouse precision could make it look like the tablet only doesn't work during gameplay.

As for your sensitivity in the tablet driver not working, that should not be an issue relevant to osu! if you at least have mouse confinement set to "never" and high precision mouse disabled, unless this is an issue with SDL itself which I highly doubt.

As for your gaomon tablet not being detected, see https://github.com/OpenTabletDriver/OpenTabletDriver/wiki/Windows-FAQ#my-gaomonhuion-tablet-isnt-detected-but-is-listed-as-supported.

**Reply by amchoudhry** [reply]
well nothing is working so idk what im suppose to do. like is there another solution to get my tablet detected?

**Reply by frenzibyte** [reply]
Did you check the link I've attached and followed the solution there? did it work for you? did it not? please refrain from posting irrelevant replies in this thread otherwise.

**Reply by amchoudhry** [reply]
yes i did and it didnt work

**Reply by frenzibyte** [reply]
Follow the last point in that FAQ then, https://github.com/OpenTabletDriver/OpenTabletDriver/wiki/Windows-FAQ#did-a-fix-not-work.

**Reply by amchoudhry** [reply]
my tablet is now detected thank you
